### DISCUSSION TITLE
Add possiblity for callbacks with `then()` and `finally()` on job chains

### DISCUSSION THREAD

**Discussion by PaulMorel** [laravel/framework#38930] [Ideas]
Hi there,

Something I noticed while working with Jobs. When using `Bus::batch()` you can setup callbacks the the `then()`, `finally()` and `catch()` methods, however it's not the case with `Bus::chain()`. From what I could gather, only `catch()` is available.

Is this intentional or merely an oversight?

I was comparing the `Illuminate\Bus\PendingBatch` and `Illuminate\Foundation\Bus\PendingChain` classes, assuming these are the correct classes.

**Comment by crynobone** [comment]
You can use Chained in Batches if you have this requirement: https://laravel.com/docs/10.x/queues#chains-within-batches
